control-plane/gateways/deployment_init_container.go

```diff
@@ -157,13 +158,11 @@ func initContainer(config GatewayConfig, name, namespace string) (corev1.Contain
 // the init container.
 // TODO @GatewayManagement parametrize gateway kind.
 const initContainerCommandTpl = `
-consul-k8s-control-plane connect-init \
-  -pod-name=${POD_NAME} \
-  -pod-namespace=${POD_NAMESPACE} \
-  -gateway-kind="mesh-gateway" \
-  -log-json={{ .LogJSON }} \
-  {{- if .AuthMethod }}
-  -service-account-name="{{ .ServiceAccountName }}" \
+consul-k8s-control-plane mesh-init \
+  -proxy-name=${POD_NAME} \
+  -namespace=${POD_NAMESPACE} \
+  {{- with .LogLevel }}
+  -log-level={{ . }} \
```

Comment:
So is `ServiceAccountName` no longer necessary or do we just not have it connected yet?

Reply:
not necessary AFAICT